### PR TITLE
cdiff 0.9.7 (new formula)

### PR DESCRIPTION
**Pull request by josa42**

### CODE REVIEW

File: Library/Formula/cdiff.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Cdiff < Formula
+  desc "View colored, incremental diff in workspace or from stdin with side by side and auto pager support"
```
Comment (DomT4) [inline Library/Formula/cdiff.rb]: This is likely to be too long. Limit is 80 characters including the formula name.

File: Library/Formula/cdiff.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  homepage "https://github.com/ymattw/cdiff"
+  head "https://github.com/ymattw/cdiff.git"
```
Comment (DomT4) [inline Library/Formula/cdiff.rb]: `head` should be the last line of the initial block here.

File: Library/Formula/cdiff.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  url "https://github.com/ymattw/cdiff/archive/0.9.7.tar.gz"
+  sha256 "41eff442efc34e545d89b94f47da22ffc5edb4224476728746841b64b69f3512"
+
+  def install
+    system "python", "./setup.py", "install", "--prefix", prefix
```
Comment (DomT4) [inline Library/Formula/cdiff.rb]: This should use something like:

```ruby
    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
    system "python", *Language::Python.setup_install_args(libexec)

    bin.install Dir["#{libexec}/bin/*"]
    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
```

File: Library/Formula/cdiff.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  end
+
+  test do
+    system "cdiff --version"
```
Comment (DomT4) [inline Library/Formula/cdiff.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Comment by josa42** [comment]
Sorry, this took a while ;-)

**Comment by josa42** [comment]
I don't get why the `brew audit` false on travis. It's seems to be fine locally.

```
$ brew audit ./Library/Formula/cdiff.rb --strict
==> brew style cdiff

1 file inspected, no offenses detected
```

**Comment by DomT4** [comment]
Travis is rate limited. Ignore it for now; the Jenkins CI is still the primary one.

**Comment by josa42** [comment]
@DomT4 Is there still something wrong with this?

File: Library/Formula/cdiff.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  sha256 "41eff442efc34e545d89b94f47da22ffc5edb4224476728746841b64b69f3512"
+  head "https://github.com/ymattw/cdiff.git"
+
+  def install
+    system "python", "./setup.py", "install", "--prefix", prefix
+  end
+
+  test do
+    file_path = test_fixtures("test.diff")
+    diff = pipe_output("cat #{file_path} | #{bin}/cdiff -cnever")
```
Comment (bfontaine) [inline Library/Formula/cdiff.rb]: Use `pipe_output("#{bin}/cdiff -cnever", File.read(file_path))`

File: Library/Formula/cdiff.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    assert_equal(diff, File.read(file_path))
```
Comment (bfontaine) [inline Library/Formula/cdiff.rb]: You can drop the parentheses here: `assert_equal diff, File.read(file_path)`

Comment (bfontaine) [inline Library/Formula/cdiff.rb]: In fact you could avoid reading twice the same file:
```
file_path = test_fixtures("test.diff")
content = File.read(file_path)
assert_equal content, pipe_output("#{bin}/cdiff -cnever", content)
```

File: Library/Formula/cdiff.rb [hunk 7]
```diff
@@ -0,0 +1,17 @@
+class Cdiff < Formula
+  desc "View colored diff with side by side and auto pager support"
+  homepage "https://github.com/ymattw/cdiff"
+  url "https://github.com/ymattw/cdiff/archive/0.9.7.tar.gz"
+  sha256 "41eff442efc34e545d89b94f47da22ffc5edb4224476728746841b64b69f3512"
+  head "https://github.com/ymattw/cdiff.git"
+
+  def install
+    system "python", "./setup.py", "install", "--prefix", prefix
```
Comment (dunn) [inline Library/Formula/cdiff.rb]: You can use our helper method here:
```ruby
system "python", *Language::Python.setup_install_args(prefix)

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

**Comment by bfontaine** [comment]
Thank you for your contribution to Homebrew @josa42; we appreciate it! :tada:

**Comment by galaxy001** [comment]
How to make homebrew python package working ?
````
cdiff
Traceback (most recent call last):
  File "/usr/local/bin/cdiff", line 5, in <module>
    import cdiff
ImportError: No module named cdiff
````

**Comment by bfontaine** [comment]
@galaxy001 Sorry the install was broken, run `brew update && brew upgrade cdiff`; it should work fine now.

**Pull request closed without merging**